src/Arcus.Observability.Telemetry.Serilog.Sinks.ApplicationInsights/Converters/ApplicationInsightsTelemetryConverter.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using Arcus.Observability.Telemetry.Core;
+using Arcus.Observability.Telemetry.Serilog.Sinks.ApplicationInsights.Converters.Dependencies;
+using Microsoft.ApplicationInsights.Channel;
+using Serilog.Events;
+using Serilog.Sinks.ApplicationInsights.Sinks.ApplicationInsights.TelemetryConverters;
+
+namespace Arcus.Observability.Telemetry.Serilog.Sinks.ApplicationInsights.Converters
+{
+    public class ApplicationInsightsTelemetryConverter : TelemetryConverterBase
+    {
+        private readonly TraceTelemetryConverter _traceTelemetryConverter = new TraceTelemetryConverter();
+        private readonly EventTelemetryConverter _eventTelemetryConverter = new EventTelemetryConverter();
+        private readonly RequestTelemetryConverter _requestTelemetryConverter = new RequestTelemetryConverter();
+
+        private readonly HttpDependencyTelemetryConverter _httpDependencyTelemetryConverter =
+            new HttpDependencyTelemetryConverter();
+
+        private readonly SqlDependencyTelemetryConverter _sqlDependencyTelemetryConverter =
+            new SqlDependencyTelemetryConverter();
+
+        public override IEnumerable<ITelemetry> Convert(LogEvent logEvent, IFormatProvider formatProvider)
+        {
+            if (logEvent.MessageTemplate.Text.StartsWith(MessagePrefixes.RequestViaHttp))
```

Comment:
Arcus code guidelines states we have braces here, no?

Reply:
I'll have to check, but switch/pattern-matching is maybe already ported from  F# to C# here.